test/parallel/test-stream-construct-async-error.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+const common = require('../common');
+const { Duplex } = require('stream');
+const { setTimeout } = require('timers/promises');
+const assert = require('assert');
+
+class Foo extends Duplex {
+  async _construct(cb) {
+    // eslint-disable-next-line no-restricted-syntax
```

Comment:
This is required because the linter does not yet recognize the single argument promisified version of `setTimeout` used this way.